### DISCUSSION TITLE
How to speed up convert function?

### DISCUSSION THREAD

**Discussion by MackonisDovydas** [ImageMagick/ImageMagick#1936] [General]
Hello,
wanted to know, if is there a way to speed up convert function by giving it more resources. Have 24 cpu's in server, but changing thread value in policy.xml do not change anything. Converting pdf to tiff, with a complex command: `convert \( -density 300 FILE.pdf -background white -strip -depth 8 -alpha remove -alpha off \) \( -clone 0 -morphology close rectangle:1x50 -negate \) \( -clone 0 -morphology close rectangle:50x1 -negate \) \( -clone 1 -clone 2 -evaluate-sequence add \) -delete 1,2 -compose plus -composite -alpha off End_FILE.tiff`
Is there a way?

**Comment by snibgo** [comment]
Changing the thread value in policy.xml can only reduce, not increase, resources available to IM. What does "convert -list policy" say?

Rasterizing the PDF is done by Ghostscript. I don't think that is multi-threaded.

You could find the time needed for each phase. Compare the time for your complete command with the times required for:
`convert -density 300 FILE.pdf x.tiff`
... and ...
`convert -density 300 FILE.pdf NULL:`

Also examine the resources used in each phase. Do you have idle CPUs during morphology etc?

Also ensure that memory usage is not a problem, which would cause pixel caches to be written to disk.

**Comment by fmw42** [comment]
If you have to do lots of those conversions, then you could restrict the threads to 1 and launch multiple processed in scripts so that each one processes one image with one thread.
